### DISCUSSION TITLE
Grid_Sampler for training

### DISCUSSION THREAD

**Discussion by MFaizyabAli** [TorchIO-project/torchio#505] [Q&A] [answered]
I wanted to ask if we can use grid sampler for training, rather than just for inference?

Does torch have any option for that?

**Comment by fepegar** [comment]
Hi, @MFaizyabAli. What would you like exactly? To use the queue and fill it using all patches provided by a grid sampler?

**Reply by MFaizyabAli** [reply]
Yes, that is exactly what I want.  I want the grid sampler to work with Queue.
Thank you!

**Reply by fepegar** [reply]
Have you tried passing a grid sampler to the Queue?

**Reply by MFaizyabAli** [reply]
Well, the grid sampler requires an instance of Subject to initialize. If I do that, it would generate patches for just one subject. Is there a way to do it over a list if subjects?

**Reply by fepegar** [reply]
You're right, the implementation is different and probably not suitable for the queue at the moment. Can you please explain a bit more what you would like to achieve? The grid sampler is used to extract all locations from an image. If you wanted to train like this, I suppose we would need a way to compute the number of patches per image and set `samples_per_volume` to that number.

**Reply by MFaizyabAli** [reply]
Yes, you are right, that is exactly what I what: to train the model on patches from entire volume, and of course we could then calculate the samples_per_volume.

**Reply by fepegar** [reply]
In that case, could you please open a new issue using the "Feature request" template?

**Reply by MFaizyabAli** [reply]
Done, thank you!

**Comment by fepegar** [comment, marked as answer]
You should be able to use the grid sampler like the others in `v0.18.37`, as the feature requested in #513 has been addressed.
